File: lib/src/enum/language.dart
enum Language {
  uzbekLatin('uz'),
  uzbekCyrillic('uz-Cyrl'),
  russian('ru'),
  kazakh('kk'),
  kyrgyz('ky'),
  tajik('tg'),
  turkmen('tk'),
  azerbaijani('az'),
  armenian('hy'),
  georgian('ka'),
  ukrainian('uk'),
  english('en'),
  ozbekCyrillic('oz');

  const Language(this.locale);

  final String locale;

  /// Convert locale string to Language enum
  static Language fromLocale(String locale) {
    return Language.values.firstWhere(
      (lang) => lang.locale.toLowerCase() == locale.toLowerCase(),
      orElse: () => Language.uzbekLatin,
    );
  }
}

File: android/src/main/kotlin/org/perfect/notifications/perfect_notifications/enum/Language.kt
package org.perfect.notifications.perfect_notifications.enum

enum class LanguageEnum(val locale: String) {
    UzbekLatin("uz"),
    UzbekCyrillic("uz-Cyrl"),
    Russian("ru"),
    Kazakh("kk"),
    Kyrgyz("ky"),
    Tajik("tg"),
    Turkmen("tk"),
    Azerbaijani("az"),
    Armenian("hy"),
    Georgian("ka"),
    Ukrainian("uk"),
    English("en"),
    OzbekCyrillic("oz");

    companion object {
        fun fromLocale(locale: String): LanguageEnum {
            return entries.firstOrNull { it.locale.equals(locale, ignoreCase = true) } ?: UzbekLatin
        }
    }
}
